src/test/java/org/jvnet/hudson/plugins/platformlabeler/PlatformDetailsTaskRedhatReleaseTest.java

```diff
@@ -0,0 +1,94 @@
+package org.jvnet.hudson.plugins.platformlabeler;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+
+import java.io.File;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.reflections.Reflections;
+import org.reflections.scanners.ResourcesScanner;
+
+@RunWith(Parameterized.class)
+public class PlatformDetailsTaskRedhatReleaseTest {
+
+  private final String redhatReleaseFileName;
+  private final String expectedName;
+  private final String expectedVersion;
+  private final String expectedArch;
+
+  public PlatformDetailsTaskRedhatReleaseTest(
+      String osReleaseFileName, String expectedName, String expectedVersion, String expectedArch) {
+    this.redhatReleaseFileName = osReleaseFileName;
+    this.expectedName = expectedName;
+    this.expectedVersion = expectedVersion;
+    this.expectedArch = expectedArch;
+  }
+
+  /**
+   * Generate test parameters for Linux os-release sample files stored as resources.
+   *
+   * @return parameter values to be tested
+   */
+  @Parameters(name = "{1}-{2}-{3}")
+  public static Collection<Object[]> generateReleaseFileNames() {
+    String packageName = PlatformDetailsTaskRedhatReleaseTest.class.getPackage().getName();
+    Reflections reflections = new Reflections(packageName, new ResourcesScanner());
+    Set<String> fileNames = reflections.getResources(Pattern.compile(".*redhat-release"));
```

Comment:
Could the matching pattern be broadened so that the existing test would process both os-release and redhat-release files?  That would reduce the duplication between this file and the other file with a few cases to handle the differences between those two files.